Update translations

wjhsf opened the pull request:
* Updated translations
  * Changed "here to help" message IDs
* Removed some hard-coded text
  * Slightly rekerjiggered how American Express tooltip is generated
* Changed checkout error from API response (not localized) to generic message (localized)
  * Will have follow-up PR to add localized API responses

# Description

<!--- A longer summary of your changes, including: a description of the issue that you’re addressing, a list of required dependencies (if applicable), and any other relevant context. -->

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

Start the server and use your preferred non-English locale.
- Find a UI with updated text. Load it. Observe the text.

Start the server with the pseudolocale.
- Bottom of the home page, observe that the text above the "Contact Us" button is localized.
- Add an item to your wishlist. In the green confirmation popup, observe that the "View" link is localized.
- View your wishlist. Observe that the "Select Options" button is localized.
- Buy an item. Before entering credit card details, hover over the security code tooltip. Observe that it is localized.
- Enter an American Express credit card. Hover over the security code tooltip. Observe that it is localized and not the same as the default text.
- Attempt to check out with no items in cart. Observe localized generic error message.

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

wjhsf committed:
Update translations.

wjhsf committed:
Compile updated translations.

wjhsf committed:
Revert inaccurate labels to here_to_help.

wjhsf committed:
Change nested translated messages to simple conditional.

wjhsf committed:
Convert 'Select Options' to translatable text.

wjhsf committed:
Convert "View" link to translatable text.

wjhsf committed:
Add translations to changelog.

wjhsf committed:
Add generic error message on checkout.

wjhsf commented on packages/template-retail-react-app/app/pages/checkout/util/checkout-context.js:
```diff
@@ -311,7 +313,12 @@ export const CheckoutProvider = ({children}) => {
                     await basket.createOrder()
                 } catch (error) {
-                    mergeState({globalError: error.message})
+                    // TODO: OCAPI implements localized error messages for some error types
```
This TODO will be done in a fast follow PR.

vmarta commented on packages/template-retail-react-app/app/pages/cart/index.jsx:
```diff
@@ -75,7 +75,10 @@ const Cart = () => {
                 action: (
                     <Button variant="link" onClick={() => navigate('/account/wishlist')}>
-                        View
+                        <FormattedMessage
+                            defaultMessage="View"
+                            id="cart.link.added_to_wishlist.view_wishlist"
+                        />
```
This change is not seen when I added an item to Wishlist from the PLP page. I think there's another component that has a similar View button.

wjhsf replied:
I've updated the individual buttons with the same hard-coded message for now. I'll extract it to a helper in the follow-up PR.

vmarta commented on packages/template-retail-react-app/app/pages/account/wishlist/partials/wishlist-primary-action.jsx:
```diff
@@ -63,7 +63,10 @@ const WishlistPrimaryAction = () => {
-                        Select Options
+                        <FormattedMessage
+                            defaultMessage="Select Options"
+                            id="wishlist_primary_action.button.select_options"
```
Great message id here btw. I don't think we've documented the id naming convention in any of the READMEs.

But the gist of it is typically a message id has the following items:
1. name of the component or page
2. UI element type ← there's actually a list of expected UI element types in some internal document (I can find it if you're curious)
3. a summary or a short phrase of the message

vmarta commented on packages/template-retail-react-app/app/pages/checkout/util/checkout-context.js:
```diff
@@ -311,7 +313,12 @@ export const CheckoutProvider = ({children}) => {
-                    mergeState({globalError: error.message})
+                    // TODO: OCAPI implements localized error messages for some error types
+                    const message = formatMessage({
+                        id: 'checkout_confirmation.message.generic_error',
```
```suggestion
                        id: 'checkout.message.generic_error',
```
Minor: when the error message shows up, we're still in checkout and has not reached the confirmation page yet (where the order submission is successful)

vmarta commented on packages/template-retail-react-app/app/translations/en-US.json:
```diff
@@ -174,6 +174,9 @@
   "cart.info.removed_from_cart": {
     "defaultMessage": "Item removed from cart"
   },
```
TODO: We'll need to send this en-US file to our translators, so they will be able to translate it to other languages.

You don't need to tell them specifically what has been added, replaced, etc. Because from what I heard, their tools are able to automatically detect the changes.

wjhsf committed:
Translate all View Wishlist links

wjhsf committed:
Change `checkout_confirmation` key to just `checkout`.

It's not a confirmation if it didn't happen!

bendvc commented:
Adding an item to your wishlist resulted in an error for me. I simple logged in, went to any category and clicked the heart on a product. 

<img width="944" alt="image" src="https://user-images.githubusercontent.com/8902581/174853239-ee0ebf01-4e2a-414d-87a9-906da0ee58d7.png">

As the error suggest, its possible that popup exists outside of the provider.

alexvuong commented on packages/template-retail-react-app/app/translations/it-IT.json:
```diff
@@ -852,7 +848,7 @@
   },
   "order_summary.label.free": {
-    "defaultMessage": "Gratis"
+    "defaultMessage": "Gratuita"
```
I noticed these translation files are getting updated in this PR. Are they auto-generated from our compiled command?

wjhsf replied:
These are manual translations provided by the localization team. The compile command simply converts the easy-to-write format (`app/translations/<locale>.json`) to an easy-to-use format (`app/translations/compiled/<locale>.json`) for the app.

wjhsf committed:
Fix rendering of localized message.

Chakra toast doesn't have access to IntlProvider,
so we need formatMessage() instead of <FormattedMessage />.

vmarta commented on packages/template-retail-react-app/app/components/recommended-products/index.jsx:
```diff
@@ -133,9 +134,13 @@ const RecommendedProducts = ({zone, recommender, products, title, shouldFetch, .
                     <Button variant="link" onClick={() => navigate('/account/wishlist')}>
-                        View
+                        {formatMessage({
+                            id: 'cart.link.added_to_wishlist.view_wishlist',
```
When you refactor this View link in the other PR, please make sure to revisit this id. I'm thinking that perhaps it would be more like a global id ("global.wishlist.link.view_wishlist" ?). Currently we have some global ids for errors and account page.

vmarta reviewed (approved):
+1 on this. Onward to the follow-up PR.
